### PR TITLE
Add new event

### PR DESCRIPTION
**Pull request by adamdossa**

### CODE REVIEW

File: contracts/tokens/SecurityToken.sol [hunk 1]
```diff
@@ -467,28 +468,39 @@ contract SecurityToken is ERC20, ReentrancyGuard, SecurityTokenStorage, IERC1594
...
-                        isForceValid = true;
-                    }
-                }
-            }
+            bytes32[] memory logName = new bytes32[](modules[TRANSFER_KEY].length);
+            bool[] memory logArchive = new bool[](modules[TRANSFER_KEY].length);
+            ITransferManager.Result[] memory logValid = new ITransferManager.Result[](modules[TRANSFER_KEY].length);
+            (unarchived, isForceValid, isInvalid, isValid) = _execute(modules[TRANSFER_KEY], _from, _to, _value, _data, logName, logArchive, logValid);
+
+            emit Restrictions(modules[TRANSFER_KEY], logName, logArchive, logValid);
```
Comment (maxsam4) [inline contracts/tokens/SecurityToken.sol]: The option to disable this is missing.

File: contracts/tokens/SecurityToken.sol [hunk 2]
```diff
@@ -467,28 +468,39 @@ contract SecurityToken is ERC20, ReentrancyGuard, SecurityTokenStorage, IERC1594
...
-                    if (valid == ITransferManager.Result.INVALID) {
-                        isInvalid = true;
-                    } else if (valid == ITransferManager.Result.VALID) {
-                        isValid = true;
-                    } else if (valid == ITransferManager.Result.FORCE_VALID) {
-                        isForceValid = true;
-                    }
-                }
-            }
+            bytes32[] memory logName = new bytes32[](modules[TRANSFER_KEY].length);
```
Comment (satyamakgec) [inline contracts/tokens/SecurityToken.sol:675]: minor optimization `uint256 length = modules[TRANSFER_KEY].length;`

File: contracts/tokens/SecurityToken.sol (hunk 1 above)
Comment (satyamakgec) [reply contracts/tokens/SecurityToken.sol]: agreed, the switch is missing

**Comment by adamdossa** [comment]
Closing as it is too complex (due to hitting contract size limitations) to implement for 3.0.

**Pull request closed without merging**